### PR TITLE
ddl :  support  `alter table xx nocache` operations switch a table into uncacheable

### LINKED ISSUE #29645: support `alter table xxx nocache` operation

### PR DESCRIPTION
**Pull request by JayL-zxl**
### What problem does this PR solve?

handle DDL schema changes when a table is set to uncachable , we use synatx` alter table xx nocache ` to switch a cacheable table to uncachable table.

Issue number: close #29645

Problem Summary: 

This is a subtask in  #25293

### What is changed and how it works?
support switch a  **cachable** table to **uncachable** operation. and do some compatibility check with other feature.
`ALTER TABLE t NOCACHE `is a DDL operation. for consistency under distributed conditions. an intermediate Switching state is introduced, the schema change process is similar to the table lock implementation:
 Enabled => Switching => Disabled
### Check List

Tests 

-  Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Review by lcwangchao — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

File: ddl/table_test.go [hunk 1]
```diff
@@ -432,6 +437,19 @@ func checkTableCacheTest(c *C, d *ddl, dbInfo *model.DBInfo, tblInfo *model.Tabl
 	c.Assert(err, IsNil)
 }
 
+func checkTableNoCacheTest(c *C, d *ddl, dbInfo *model.DBInfo, tblInfo *model.TableInfo) {
+	err := kv.RunInNewTxn(context.Background(), d.store, false, func(ctx context.Context, txn kv.Transaction) error {
+		t := meta.NewMeta(txn)
+		info, err := t.GetTable(dbInfo.ID, tblInfo.ID)
+		c.Assert(err, IsNil)
+		c.Assert(info, NotNil)
+		c.Assert(info.TableCacheStatusType, NotNil)
```
Comment (tangenta) [inline ddl/table_test.go:446]:
```suggestion
```

Comment (tangenta) [reply ddl/table_test.go:446]: There is no way to be `nil` for an integer type.

File: ddl/db_cache_test.go [hunk 2]
```diff
@@ -110,3 +111,26 @@ func (s *testDBSuite2) TestAlterViewTableCache(c *C) {
 	tk.MustExec("create view v as select * from cache_view_t")
 	tk.MustGetErrCode("alter table v cache", errno.ErrWrongObject)
 }
+
+func (s *testDBSuite2) TestAlterTableNoCache(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk2 := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists nocache_t1")
+	tk2.MustExec("use test")
```
Comment (hawkingrei) [inline ddl/db_cache_test.go:120]: tk2 just do one thing？

Comment (JayL-zxl) [reply ddl/db_cache_test.go:120]: sorry. don't need tk2 here.

**Review by tangenta — APPROVED** [review]

**Comment by JayL-zxl** [comment]
My pr encountered ci. Already resolved in this pr#29443

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**